test/test_um2netcdf.py

```diff
@@ -651,6 +662,379 @@ def test_fix_units_do_nothing_no_um_units(ua_plev_cube):
+
+    assert_dtype_float64(cube_lat_coord, cube_lon_coord)
+    assert_has_bounds(cube_lat_coord, cube_lon_coord)
+
+
+def test_fix_latlon_coords_uv(ua_plev_cube,
+                              lat_v_nd,
+                              lon_u_nd,
+                              lat_v_eg,
+                              lon_u_eg):
+    """
+    Tests of the fix_lat_lon_coords for longitude u and latitude v
+    coordinates on both the New Dynamics and End Game grids.
+    """
+    coord_sets = [
+        (lat_v_nd, lon_u_nd, um2nc.GRID_NEW_DYNAMICS),
+        (lat_v_eg, lon_u_eg, um2nc.GRID_END_GAME)
+    ]
+
+    for lat_coordinate, lon_coordinate, grid_type in coord_sets:
+
+        cube_with_uv_coords = DummyCubeWithCoords(
+            dummy_cube=ua_plev_cube,
+            coords=[lat_coordinate, lon_coordinate]
+        )
+
+        # Checks prior to modifications.
+        assert_unmodified_coordinates(lat_coordinate, lon_coordinate)
+
+        um2nc.fix_latlon_coords(cube_with_uv_coords, grid_type,
+                                D_LAT_N96, D_LON_N96)
+
+        assert lat_coordinate.var_name == um2nc.VAR_NAME_LAT_V
+        assert lon_coordinate.var_name == um2nc.VAR_NAME_LON_U
+
+        assert_dtype_float64(lat_coordinate, lon_coordinate)
+        assert_has_bounds(lat_coordinate, lon_coordinate)
+
+
+def test_fix_latlon_coords_standard(ua_plev_cube,
+                                    lat_standard_nd,
+                                    lon_standard_nd,
+                                    lat_standard_eg,
+                                    lon_standard_eg):
+    """
+    Tests of the fix_lat_lon_coords for standard longitude
+    and latitude coordinates on both the New Dynamics and
+    End Game grids.
+    """
+    coord_sets = [
+        (
+            lat_standard_nd,
+            lon_standard_nd,
+            um2nc.GRID_NEW_DYNAMICS
+         ),
+        (
+            lat_standard_eg,
+            lon_standard_eg,
+            um2nc.GRID_END_GAME
+        )
+    ]
+
+    for lat_coordinate, lon_coordinate, grid_type in coord_sets:
+
+        cube_with_uv_coords = DummyCubeWithCoords(
+            dummy_cube=ua_plev_cube,
+            coords=[lat_coordinate, lon_coordinate]
+        )
+
+        # Checks prior to modifications.
+        assert_unmodified_coordinates(lat_coordinate, lon_coordinate)
+
+        um2nc.fix_latlon_coords(cube_with_uv_coords, grid_type,
+                                D_LAT_N96, D_LON_N96)
+
+        assert lat_coordinate.var_name == um2nc.VAR_NAME_LAT_STANDARD
+        assert lon_coordinate.var_name == um2nc.VAR_NAME_LON_STANDARD
+
+        assert_dtype_float64(lat_coordinate, lon_coordinate)
+        assert_has_bounds(lat_coordinate, lon_coordinate)
+
+
+def test_fix_latlon_coords_single_point(ua_plev_cube):
+    """
+    Test that single point longitude and latitude coordinates
+    are provided with global bounds.
+    """
+    grid_type = um2nc.GRID_NEW_DYNAMICS
```

Comment:
Minor thing, line 976 can use the constant directly.

Reply:
Good idea, using the constants directly now